src/main/java/tech/andreagreco/AnotherExample.java

```diff
@@ -0,0 +1,36 @@
+package tech.andreagreco;
+
+import java.math.BigDecimal;
+import java.util.Deque;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class AnotherExample {
+    
+    public void function() {
+        final ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+        final Deque<BigDecimal> numbers = new LinkedBlockingDeque<>();
+        final BigDecimal divisor = new BigDecimal(51);
+
+        scheduledExecutorService.scheduleAtFixedRate(() -> {
```

Comment:
Not clear what this is doing, can you extract this function and give clear names to variables that explain what this is doing?